pkg/topology/test/helper.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package test
+
+import (
+	"fmt"
+	"math/rand"
+
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+// RandomAddressAt generates a random address
+// at proximity order prox relative to address
+// if prox is negative a random address is generated.
```

Comment:
I would conceal this with a `randomAddressAt`, in order to not overload the functionality of `RandomAddress`

Reply:
not sure if i follow. wdym?

Reply:
Remove the negative option on this method. If you want fully random only call the other.

```
randomAddress(address a, int x) : x € Z

RandomAddressAt(address a, int x) : x >= 0
* calls randomAddressAt(a, x)
RandomAddress()
* calls randomAddressAt(a, -1)
```

Reply:
so you end up with three functions that do exactly what these two functions do, and your `RandomAddressAt` still accepts an `int` meaning it can take a negative value, but the condition you mention here does not handle that case. so if it is negative is the function supposed to panic? i dont really understand...

Reply:
It seems to me in general that in our code and in `golang` overloading a method argument's (or return value's) with special meaning for negative numbers is not standard.

Perhaps just remove the comment line for the negative value?

Reply:
> It seems to me in general that in our code and in `golang` overloading a method argument's (or return value's) with special meaning for negative numbers is not standard.

That's not entirely correct. See [1](https://golang.org/pkg/strings/#Index) and [2](https://golang.org/pkg/strings/#Replace)

It is not common, but it is definitely something acceptable.

I'll remove the comment about the negative number. Thanks